src/Controls/src/Core/BindingExpression.cs

```diff
@@ -427,6 +427,22 @@ void SetupPart(TypeInfo sourceType, BindingExpressionPart part)
 		}
 
 		static readonly Type[] DecimalTypes = { typeof(float), typeof(decimal), typeof(double) };
+		static readonly Type[] NumericTypes =
```

Comment:
This may be simplified using generic math, type checking for `INumberBase` and using the `Zero` property. Please see

https://learn.microsoft.com/en-us/dotnet/api/system.numerics.inumberbase-1?view=net-8.0